server/src/test/java/org/elasticsearch/transport/InboundHandlerTests.java

```diff
@@ -256,6 +255,43 @@ public void testClosesChannelOnErrorInHandshakeWithIncompatibleVersion() throws
         }
     }
 
+    public void testLogsSlowInboundProcessing() throws Exception {
+        final MockLogAppender mockAppender = new MockLogAppender();
+        mockAppender.start();
+        mockAppender.addExpectation(
+            new MockLogAppender.SeenEventExpectation(
+                "expected message",
+                InboundHandler.class.getCanonicalName(),
+                Level.WARN,
+                "handling inbound transport message "));
+        final Logger inboundHandlerLogger = LogManager.getLogger(InboundHandler.class);
+        Loggers.addAppender(inboundHandlerLogger, mockAppender);
+
+        handler.setSlowLogThreshold(TimeValue.timeValueMillis(5L));
+        try {
+            final Version remoteVersion = Version.CURRENT;
+            final long requestId = randomNonNegativeLong();
+            final Header requestHeader = new Header(between(0, 100), requestId,
+                TransportStatus.setRequest(TransportStatus.setHandshake((byte) 0)), remoteVersion);
+            final InboundMessage requestMessage =
+                new InboundMessage(requestHeader, ReleasableBytesReference.wrap(BytesArray.EMPTY), () -> {
+                    try {
+                        TimeUnit.SECONDS.sleep(1L);
+                    } catch (InterruptedException e) {
+                        throw new AssertionError(e);
+                    }
+                });
+            requestHeader.actionName = TransportHandshaker.HANDSHAKE_ACTION_NAME;
+            requestHeader.bwcNeedsToReadVariableHeader = false;
```

Comment:
this is reflecting headers provision in https://github.com/elastic/elasticsearch/commit/5112c173194b70d2ed17f2ed3071dc0c9a44b76e#diff-b5518bc3e3b5862afbc9f1978088945d8c71427f421e9a703e085931ffcac577R272

We don't have `headers` Set, we have a flag.
See https://github.com/crate/crate/commit/8fbec5d8d386e5804080356020cb573cadbac678#diff-0eef2442af29607a3a1ed7df397bd704179d41d84fc0a135fe96b814513a5464R88 and 
https://github.com/elastic/elasticsearch/commit/2ccddbfa88e#diff-0eef2442af29607a3a1ed7df397bd704179d41d84fc0a135fe96b814513a5464R95

Reply:
Maybe add a comment on top of the flag set to explain?

Reply:
Added comment in https://github.com/crate/crate/pull/13409/commits/29a3516b5d1397a05fca7c4b4ed2cd9a186e4b00